### PR TITLE
Fix string length limiting for double-width characters

### PR DESCRIPTION
**Pull request by RobertKwiatkowski**
https://github.com/Evertras/bubble-table/issues/40
Implemented the go-runewidth library to manage runes truncation
Added tests for japanese characters

### CODE REVIEW

File: table/strlimit.go [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+
 func limitStr(str string, maxLen int) string {
 	if maxLen == 0 {
 		return ""
 	}
-
-	if len([]rune(str)) > maxLen {
-		return string([]rune(str)[:maxLen-1]) + "…"
+	if runewidth.StringWidth(str) > maxLen {
+		return runewidth.Truncate(str, maxLen-1, "") + "…" // before str[:maxLen] + "…"
```
Comment (Evertras) [inline table/strlimit.go:10]:
```suggestion
		return runewidth.Truncate(str, maxLen-1, "") + "…"
```

File: examples/flex/main.go [hunk 2]
```diff
@@ -36,10 +36,8 @@ func NewModel() Model {
 				columnKeyDescription: "Super zappy mouse, handle with care",
 			}),
 			table.NewRow(table.RowData{
-				columnKeyName:    "Charmander",
-				columnKeyElement: "Fire",
-				// TODO: Fix double width string length limiting!
-				//columnKeyDescription: "直立した恐竜のような身体と、尻尾の先端に常に燃えている炎が特徴。",
+				columnKeyName:        "Charmander",
+				columnKeyElement:     "Fire",
 				columnKeyDescription: "Lots of fire",
```
Comment (Evertras) [inline examples/flex/main.go:41]:
```suggestion
				columnKeyDescription: "直立した恐竜のような身体と、尻尾の先端に常に燃えている炎が特徴。",
```

I'd like to keep this example in place to show how it truncates wide characters.

**Comment by Evertras** [comment]
You can run `make lint` to check linting locally.  It looks like it's whining about a function length and I think this is a false positive, because this is just adding straightforward test cases.  To get around this, you'll want to add comments like in https://github.com/Evertras/bubble-table/blob/main/table/dimensions_test.go#L10-L11

```golang
// This function is only long because of repetitive test definitions, this is fine
// nolint: funlen
```

Put that immediately in front of the test function and the linting should then pass.

File: table/strlimit.go (hunk 1 above)
Comment (RobertKwiatkowski) [reply table/strlimit.go]: My bad, I forgot to delete it because I used it during development to take a look at the original truncate.

**Comment by RobertKwiatkowski** [comment]
Sorry, wait a second let me change commit author because accidentally used my company one.

**Comment by Evertras** [comment]
> Unfortunately I can't due to usage of windows. I guess it's high time to use the WSL 😛

Ahh, yeah, this is built on both a Mac and a Linux machine, no Windows.  Definitely recommend WSL in general for the general dev ecosystem.

And no worries, I also just noticed the commit author was slightly different than your user.  LGTM though, will merge once you give the clear.

**Comment by Evertras** [comment]
LGTM, thanks for the contribution!

**Comment by RobertKwiatkowski** [comment]
Well, it's easy to forget about the global git config...

Commit&author looks goods now, feel free to merge it.

Btw. It was really nice to cooperate with you, thank for all the insights :).

**Pull request merged**